lib/one_model.ex

```diff
@@ -161,6 +161,9 @@ defmodule OneModel do
       * `select: atom | list`
         * Passing an atom will return the value of the given atom key
         * Passing a list of keys will return the default schema with the values for the specified keys.
+      * `select_map: atom | list`
```

Comment:
I've got a different proposal. Rather than creating a `:select_map` option, let's expand upon the current `:select` option.
1.  Have it accept a map of fields of the form `%{field: 1, another_field: 1, some_other_field: 1}`. 
    - The map of fields to select would be consistent in format to the way query_sort_and_paginate fields are specified.
    - It could be called as such `list(select: :maps.from_keys([:a, :b], 1))`. 
    -  Could be extended to support fields to not select.
2. Similar to the above, but instead of map, accept a tuple of the type `{[atom], non_neg_integer}` where the integer is only either 0 or 1.
    - I only include this in case the first suggestion is too difficult to easily determine if the operation should be a select or reject. Although, I don't think it should be since we could more than likely reuse the logic from the query_sort_and_paginate selection.

Reply:
@palexanderm As discussed: 

* The API query fields is an external API that makes sense. This PR is about an intenal API to be used by our code. They ary not the same thing.
* I can't think of a use case returning a map with all keys except those provede in the map `%{field1: 0, field2: 0}`. 
* Given that we will probably never use that option, `[:field1, :field2]` is more concise than `%{field1: 1, field2: 1}`. Furthermore, it aligns well with ecto, we just need to tell ecto if we want the struct returned (with default values) or a map. 
* I consided adding an option to the select key, but that just complicates select. A separate option is cleaner. 

So, I'll continue with the current design, but  remove the clause that acccepts an atom. So that will raise with a function clause error.